zebra-chain/src/sprout/joinsplit.rs

```diff
@@ -91,8 +92,8 @@ impl<P: ZkSnarkProof> ZcashSerialize for JoinSplit<P> {
 impl<P: ZkSnarkProof> ZcashDeserialize for JoinSplit<P> {
     fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
         Ok(JoinSplit::<P> {
-            vpub_old: reader.read_u64::<LittleEndian>()?.try_into()?,
-            vpub_new: reader.read_u64::<LittleEndian>()?.try_into()?,
```

Comment:
are these supposed to read the amount as a `u64`?

Reply:
Yes, spec 7.2 Encoding of JoinSplit Descriptions says:

Name | Data Type | Description
--- | --- | ---
vpub_old | uint64 | A value that the JoinSplit transfer removes from the transparent transaction value pool.
vpub_new | uint64 | A value that the JoinSplit transfer inserts into the transparent transaction value pool.